### PR TITLE
Update test pipelines to be triggered by release/1.2

### PR DESCRIPTION
**Pull request by yophilav**
Update test pipelines to be triggered by release/1.2

### CODE REVIEW

File: builds/checkin/api-proxy.yaml [hunk 1]
```diff
@@ -3,7 +3,14 @@ pr:
   branches:
     include:
       - master
-      - iiot
+      - release/*
+  paths:
+    include:
+      - "builds/edge-modules/api-proxy-module/*"
```
Comment (damonbarry) [inline builds/checkin/api-proxy.yaml:9]: `builds/edge-modules/...` isn't a valid path. I think you just want `edge-modules/api-proxy-module`.

**Review by damonbarry — CHANGES_REQUESTED** [review]

File: builds/checkin/api-proxy.yaml [hunk 2]
```diff
@@ -3,7 +3,14 @@ pr:
   branches:
     include:
       - master
-      - iiot
+      - release/*
+  paths:
+    include:
+      - "edge-modules/api-proxy-module/*"
```
Comment (damonbarry) [inline builds/checkin/api-proxy.yaml:9]: Looking at this a little more carefully....I don't think we want to include a `paths` element to the `pr` spec here, because the `check_run_pipeline` job is already doing that work.

(`paths` is a much simpler, cleaner way to restrict when this pipleline runs, but we added the `check_run_pipeline` job to our CI builds as a workaround for some problem related to kodiak--I don't remember the details.)

**Review by damonbarry — CHANGES_REQUESTED** [review]

File: builds/checkin/api-proxy.yaml (hunk 2 above)
Comment (vadim-kovalyov) [reply builds/checkin/api-proxy.yaml:9]: Correct, having path filter does not work, I tried that before and failed :) The problem is that we have several required checks on PRs in GH, and if a required pipeline does not start (and it will not start if path filter does not match), no status is reported back to GH from ADO, and the PR is forever stuck in "waiting for pipeline" status

File: builds/checkin/api-proxy.yaml [hunk 3]
```diff
@@ -3,7 +3,14 @@ pr:
...
+
+variables:
```
Comment (vadim-kovalyov) [inline builds/checkin/api-proxy.yaml:11]: I don't think we need this for PR checks. It is only needed for internal (msazure) pipelines.

File: builds/e2e/e2e.yaml [hunk 4]
```diff
@@ -8,14 +8,14 @@ resources:
   pipelines:
   - pipeline: images
     source: 'Azure-IoT-Edge-Core Build Images'
-    branch: master
+    branch: 'release/1.2'
```
Comment (vadim-kovalyov) [inline builds/e2e/e2e.yaml:11]: can I ask why this change? that is where the pipeline definition is coming from, not the trigger bransh, AFAIK

File: builds/misc/images-mqtt.yaml [hunk 5]
```diff
@@ -3,6 +3,7 @@ trigger:
   branches:
     include:
       - master
+      - release/*
```
Comment (vadim-kovalyov) [inline builds/misc/images-mqtt.yaml:6]: nit: that won't hurt, but not really necessary. This pipeline (and image) is used for MQTT perf tests. And we are fine testing master only.

File: builds/e2e/e2e.yaml (hunk 4 above)
Comment (yophilav) [reply builds/e2e/e2e.yaml:11]: Yeah. I think we want to this pipeline to be using yaml from this branch right?

File: builds/misc/images-mqtt.yaml (hunk 5 above)
Comment (yophilav) [reply builds/misc/images-mqtt.yaml:6]: Do you think it would be a reasonable thing to do to have this pipeline be run as well? If not, I'll just remove it.

File: builds/e2e/e2e.yaml (hunk 4 above)
Comment (damonbarry) [reply builds/e2e/e2e.yaml:11]: This isn't the trigger branch, this is a filter on grabbing artifacts from "Azure-IoT-Edge-Core Build Images" builds. So this will cause the end-to-end test pipeline to use artifacts from the latest successful "Build Images" build _against the release/1.2 branch_.

File: builds/misc/images-mqtt.yaml (hunk 5 above)
Comment (vadim-kovalyov) [reply builds/misc/images-mqtt.yaml:6]: There is no use for now to run it on release. We don't ship any artifacts produced by this pipeline.

File: builds/e2e/e2e.yaml (hunk 4 above)
Comment (yophilav) [reply builds/e2e/e2e.yaml:11]: Thank you for the clarification! I think this is what we want here.

**Review by damonbarry — COMMENTED** [review]
LGTM, but I think there's still one comment from @vadim-kovalyov to address.

**Review by vadim-kovalyov — APPROVED** [review]
🚢
